src/PowerShellEditorServices/Services/Symbols/SymbolsService.cs

```diff
@@ -408,123 +299,56 @@ static string[] GetIdentifiers(string symbolName, SymbolType symbolType, Command
-                    if (scriptFile.FilePath == dotSourcedPath)
+                    if (foundReference.IsDeclaration)
                     {
-                        foundDefinition = new SymbolReference(SymbolType.Function, foundSymbol.SymbolName, scriptFile.ScriptAst.Extent, scriptFile.FilePath);
-                        break;
+                        _logger.LogDebug($"Found possible declaration in same file ${foundReference}");
+                        declarations.Add(foundReference);
                     }
                 }
             }
 
-            // if the definition the not found in referenced files
-            // look for it in all the files in the workspace
-            if (foundDefinition is null)
+            if (declarations.Any())
             {
-                // Get a list of all powershell files in the workspace path
-                foreach (string file in _workspaceService.EnumeratePSFiles())
-                {
-                    if (filesSearched.Contains(file))
-                    {
-                        continue;
-                    }
-
-                    foundDefinition =
-                        AstOperations.FindDefinitionOfSymbol(
-                            Parser.ParseFile(file, out Token[] tokens, out ParseError[] parseErrors),
-                            foundSymbol);
-
-                    filesSearched.Add(file);
-                    if (foundDefinition is not null)
-                    {
-                        foundDefinition.FilePath = file;
-                        break;
-                    }
-                }
+                return declarations;
             }
 
-            // if the definition is not found in a file in the workspace
-            // look for it in the builtin commands but only if the symbol
-            // we are looking at is possibly a Function.
-            if (foundDefinition is null
-                && (foundSymbol.SymbolType == SymbolType.Function
-                    || foundSymbol.SymbolType == SymbolType.Unknown))
+            foreach (SymbolReference foundReference in await ScanForReferencesOfSymbolAsync(
+                    symbol, cancellationToken).ConfigureAwait(false))
             {
-                CommandInfo cmdInfo =
-                    await CommandHelpers.GetCommandInfoAsync(
-                        foundSymbol.SymbolName,
-                        _runspaceContext.CurrentRunspace,
-                        _executionService).ConfigureAwait(false);
-
-                foundDefinition =
-                    FindDeclarationForBuiltinCommand(
-                        cmdInfo,
-                        foundSymbol);
+                if (foundReference.IsDeclaration)
+                {
+                    _logger.LogDebug($"Found possible declaration in workspace ${foundReference}");
+                    declarations.Add(foundReference);
+                }
             }
 
-            return foundDefinition;
+            return declarations;
```

Comment:
```suggestion
            return declarations.Where((d) =>
                    symbol.SymbolType == d.SymbolType ||
                    (symbol.SymbolType is SymbolType.Class or SymbolType.Enum or SymbolType.Type
                        && d.SymbolType is SymbolType.Class or SymbolType.Enum) ||
                    (symbol.SymbolType is SymbolType.EnumMember or SymbolType.Property
                        && d.SymbolType is SymbolType.EnumMember)
                );
```

Fix go to definition symbol type mismatch (part 2)